### ISSUE TITLE
ERROR: client is newer than server (client API version: 1.22, server API version: 1.21)

### ISSUE THREAD

**Issue by christianbundy** [docker/machine#3190]
Docker Machine can currently create a VM, but running `docker-compose build` will fail with the above error.

**Comment by nathanleclaire** [comment]
Hi @christianbundy, thanks for the issue.

Please update your `docker-compose` version through the [Toolbox](https://github.com/docker/toolbox) for API compatibility.

**Comment by christianbundy** [comment]
Thanks @nathanleclaire, maybe I should've been more clear: the problem is that the generic driver comes with an old version of Docker installed by default, which seems like a bug. This is happening with all `create` commands with docker-machine version 0.6.0, build e27fb87. For reference, this was the create command used:

``` shell
docker-machine create \
--driver amazonec2 \
--amazonec2-access-key $AWS_ACCESS_KEY \
--amazonec2-instance-type t2.large \
--amazonec2-monitoring \
--amazonec2-root-size 32 \
--amazonec2-secret-key $AWS_SECRET_KEY \
--amazonec2-use-private-address \
--amazonec2-vpc-id $AWS_VPC_ID \
--amazonec2-zone $AWS_ZONE \
$1
```

Is this issue worth re-opening, or is the old version of Docker considered expected behavior? Please let me know if there's anything else I can do to make diagnosing this easier. Cheers!

**Comment by nathanleclaire** [comment]
@christianbundy No, it should always be the latest version available (1.10.3 today).  We always call `get.docker.com` to run the install, which should always give the latest version.  Are you sure you're not using `--amazonec2-ami` with a pre-existing Docker version installed? That's one reason the Docker version might be off. 

Please post a https://gist.github.com with the output of invoking that command using `--debug` so we can take a look (make sure to strip any sensitive information such as certificates).
